std/traits.d

```diff
@@ -2518,6 +2519,69 @@ private:
     }
 }
 
+/**
+ * Get an alias to a specific overload of a method.
+ *
+ * Params:
+ *    overloadSet = alias to the method's overload set
+ *    attrs = (optional) $(LREF FunctionAttribute)s of the overload to be aliased.
+ *        This is matched exactly! Note that one of @system, @trusted and @safe
+ *        is always there. Defaults to `FunctionAttribute.system`.
+ *    ParamTypes = Types of the parameters of the overload to be aliased.
+ */
+template getOverload(alias overloadSet, FunctionAttribute attrs, ParamTypes...)
+{
+    import std.meta : Filter;
+
+    private enum bool isMatch(alias fun) = (functionAttributes!fun == attrs
+        && is(ParamTypes == Parameters!fun));
+
+    private alias matches = Filter!(isMatch, __traits(getOverloads,
+        __traits(parent, overloadSet), __traits(identifier, overloadSet)));
+
+    static assert(matches.length <= 1, "Multiple overloads seem to match...");
+
+    static if(matches.length != 0)
+        alias getOverload = matches[0];
+    else
+    {
+        pragma(msg, "No matching overload found for " ~
+            __traits(identifier, overloadSet) ~ ParamTypes.stringof);
```

Comment:
Please remove the pragma msg (as an aside the preferred form of pragma message and mixin is the comma separated list form: `pragma(msg, "No matching", __traits(identifier, overloadSet), ParamTypes);`

Reply:
I'd suggest either `alias getOverload = void;` or `static assert(0, "No matching....");`when there's no match.